src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java

```diff
@@ -274,19 +279,32 @@ public void runAppCfgCommand(List<String> args) throws ProcessRunnerException {
   }
 
   /**
-   * Returns the version of the Cloud SDK installation. Unlike other methods in this class that call
-   * gcloud, this method always uses a synchronous ProcessRunner and will block until the gcloud
-   * process returns.
+   * Returns the version of the Cloud SDK installation. Version is determined by reading the VERSION
+   * file located in the Cloud SDK directory.
    *
-   * @throws ProcessRunnerException when process runner encounters an error
+   * @throws CloudSdkOutOfDateException if the Cloud SDK is too out of date to determine its version
+   * @throws IOException if the VERSION file could not be read. This might mean the Cloud SDK
+   *                     installation is corrupted.
    */
-  public CloudSdkVersion getVersion() throws ProcessRunnerException {
-    // gcloud info --format="value(basic.version)"
-    List<String> command = new ImmutableList.Builder<String>().add("info")
-        .addAll(GcloudArgs.get("format", "value(basic.version)")).build();
+  public CloudSdkVersion getVersion() throws IOException {
+    Path versionFile = getSdkPath().resolve(VERSION_FILE);
 
-    String output = runSynchronousGcloudCommand(command);
-    return new CloudSdkVersion(output);
+    if (!Files.isRegularFile(versionFile)) {
+      throw new CloudSdkOutOfDateException(MINIMUM_VERSION);
+    }
+
+    String contents = "";
+    List<String> lines = Files.readAllLines(versionFile, StandardCharsets.UTF_8);
+    if (lines.size() > 0) {
+      // expect only a single line
+      contents = lines.get(0);
```

Comment:
If `lines.size() == 0` (or could do `!lines.isEmpty()`), it will reach`return new CloudSdkVersion(contents);`. If `contents` is am empty string, the `Precondition` check inside will throw `IllegalArgumentException`.

Maybe it's simpler to catch `IndexOutOfBoundsException` in the following `try` block?